benchmarks/src/test/java/org/apache/druid/benchmark/ProtobufParserBenchmark.java

```diff
@@ -0,0 +1,173 @@
+                        new StringDimensionSchema("event"),
+                        new StringDimensionSchema("id"),
+                        new StringDimensionSchema("someOtherId"),
+                        new StringDimensionSchema("isValid")
+                ), null, null),
+                new JSONPathSpec(
+                        true,
+                        Lists.newArrayList(
+                                new JSONPathFieldSpec(JSONPathFieldType.ROOT, "eventType", "eventType"),
+                                new JSONPathFieldSpec(JSONPathFieldType.PATH, "foobar", "$.foo.bar"),
+                                new JSONPathFieldSpec(JSONPathFieldType.PATH, "bar0", "$.bar[0].bar")
+                        )
+                ),
+                null,
+                null
+    );
+
+    flattenParseSpec = new JSONParseSpec(
+            new TimestampSpec("timestamp", "iso", null),
+            new DimensionsSpec(Lists.newArrayList(
+                    new StringDimensionSchema("event"),
+                    new StringDimensionSchema("id"),
+                    new StringDimensionSchema("someOtherId"),
+                    new StringDimensionSchema("isValid")
+            ), null, null),
+            null,
+            null,
+            null
+    );
+
+    protoFilePath = "ProtoFile";
+    protoInputs = getProtoInputs(protoFilePath);
+    nestedParser = new ProtobufInputRowParser(nestedParseSpec, "prototest.desc", "ProtoTestEvent");
+    flattenParser = new ProtobufInputRowParser(flattenParseSpec, "prototest.desc", "ProtoTestEvent");
+  }
+
+  @Benchmark
+  @BenchmarkMode(Mode.AverageTime)
+  @OutputTimeUnit(TimeUnit.MICROSECONDS)
+  public void consumeFlattenData(Blackhole blackhole)
+  {
+    for (int i = 0; i < rowsPerSegment; i++) {
+      InputRow row = flattenParser.parseBatch(ByteBuffer.wrap(protoInputs)).get(0);
+      blackhole.consume(row);
+    }
+  }
+
+  @Benchmark
+  @BenchmarkMode(Mode.AverageTime)
+  @OutputTimeUnit(TimeUnit.MICROSECONDS)
+  public void consumeNestedData(Blackhole blackhole)
+  {
+    for (int i = 0; i < rowsPerSegment; i++) {
+      InputRow row = nestedParser.parseBatch(ByteBuffer.wrap(protoInputs)).get(0);
+      blackhole.consume(row);
+    }
+
+  }
+  private byte[] getProtoInputs(String fileName)
+  {
+    String filePath = this.getClass().getClassLoader().getResource(fileName).getPath();
+    byte[] bytes = null;
+    try {
+      File file = new File(filePath);
+      bytes = new byte[(int) file.length()];
+      bytes = Files.toByteArray(file);
+    }
+    catch (FileNotFoundException e) {
+      log.error("Cannot find the file: " + filePath);
+      e.printStackTrace();
+    }
+    catch (IOException e) {
+      e.printStackTrace();
+    }
+    return bytes;
+  }
+
+  public static void main(String[] args) throws RunnerException
+  {
+    Options opt = new OptionsBuilder()
+        .include(ProtobufParserBenchmark.class.getSimpleName())
+        .build();
+    new Runner(opt).run();
```

Comment:
I don't think the benchmark class should have a main method. It seems useful for debugging, but I don't think it should exist in the master branch.

Reply:
Yes you're right~ It is just for debugging and should be removed.